docs/Secure-Coding-Guide-for-Python/CWE-707/CWE-89/README.md

```diff
@@ -0,0 +1,300 @@
+# CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')
+
+To prevent SQL injections, use input sanitization and parameterized queries instead of `executescript()`.
+
+Combining an SQL query with data from lesser trusted sources, such as fields from a web form used to sign up new clients, can allow an attacker to inject malicious SQL statements.
+
+SQLs that allow running operating system commands as part of an SQL query are sensitive to Remote Code Execution (RCE) attacks such as `\system` or `\!`, in *MySQL Shell commands* [[Oracle 2024]](https://dev.mysql.com/doc/mysql-shell/8.0/en/mysql-shell-commands.html). SQL injections have been known for decades *ODBC and MS SQL server 6.5* [[Jeff Forristal 1998]](http://phrack.org/issues/54/8.html#article) and are still part of the most frequent top 10 attacks in CWE top 25 [[MITRE 2024]](https://cwe.mitre.org/top25/archive/2024/2024_cwe_top25.html) and 2021 [[OWASP 2021]](https://owasp.org/www-project-top-ten/).
+
+Note that "Raw SQL is error prone, more labor intensive, and ugly." [Byrne 2021]
+
+![Exploits of a mom](image01.webp "Exploits of a mom")
+
+Suppose a school has a webpage that allows parents to add the name of their child to a school trip themselves. The webpage is using an SQL back-end adding a new student `<NAME>` as follows:
+
+```SQL
+INSERT INTO Students(student) VALUES('<NAME>')
+```
+
+Suppose an attacker can substitute arbitrary strings for <NAME> and is allowed to extend it with any character
+
+```SQL
+<NAME>'); DROP TABLE students;--")
+```
+
+The attacker can then end and close the query himself with '); and extend the query with other SQL commands.
+
+```SQL
+INSERT INTO Students(student) VALUES('Robert'); DROP TABLE students;--');
+```
+
+`DROP TABLE students;` will delete the existing table with student names.
+
+The `help(sqlite3)` indicates that `sqlite3.cursor.executescript()` should not be used by marking it as "Non-standard" while `cursor` is considered "standard".
+
+```man
+|  cursor(...)
+|      Return a cursor for the connection.
+ 
+|  executescript(...)
+|      Executes a multiple SQL statements at once. Non-standard.
```

Comment:
See comment above. This example does not list the "standard" option.

Reply:
See previous comment, I will remove this.